# Домашнее задание к занятию "Поиск и исправление ошибок приложения. Отладка - рефакторинг"

* При сдаче домашнего задания в поле для сдачи работы прикрепите ссылку на Pull Request с кодом на проверку внутри вашего проекта в Git.
* Для сдачи обязательны все задачи, кроме задачи со звездочкой.
* Любые вопросы по решению задач задавайте в чате Slack (ссылку вы найдете в письме на вашей эл. почте).


## Задание 1
1. Найти в вашем приложении места, где есть возможность неблагоприятного исхода и требуется сообщить пользователю об этом. 
2. Создайте собственный домен ошибок (перечисление). 
3. Добавьте в него случаи, которые могут быть. 
4. Покажите пользователю Alert, уведомив его об ошибке.

## Задание 2
Примените тип `Result` для асинхронного вызова, обработайте результат через `switch`.

## Задание 3
Переделайте один из методов, который возвращает `nil` в неблагоприятном исходе, используя функцию, которая выбрасывает исключение в этом случае (`throws`).

## Задание 4* (со звездочкой)
Найдите место в вашей программе, где дальнейшее ее выполнение не имеет смысла, если определенное условие не выполнено, воспользуйтесь `preconditionFailure` и `guard`.
